onix-bot/OnixParser.py

```diff
@@ -0,0 +1,164 @@
+import onixcheck
+from lxml import etree
+
+from olclient.openlibrary import OpenLibrary
+import olclient.common as common
+import string
+import ast
+import requests
+import json
+
+FILE = 'data/SampleONIX.xml' 
+ol = OpenLibrary()
+
+class OnixParser(object):
+    """
+    ONIX Parser which parses through an ONIX File and returns valid attributes
+    Attributes:
+        products: An xml object representing all the Products in the ONIX File  
+    """
+
+    def __init__(self, filename):
+        """
+        1. Checks if there are any errors in the ONIX File 
+        2. Returns an etree object called products which allows you to parse the file
+        """
+        errors = onixcheck.validate(filename)
+        for error in errors:
+            print(error.short)
+
+        if errors:
+            print("Your ONIX Record has a mistake. Kindly check the ONIX File again before parsing.")
+            exit(0)
+
+        tree = etree.parse(filename)
+
+        self.products = tree.xpath('/ONIXMessage/Product')
+        self.onix_records = [[]]
+
+    def parse_product(self, product):
+        product = etree.fromstring(etree.tostring(product))
+        identifiers = product.xpath('/Product/ProductIdentifier')
+
+        isbn10 = None
+        isbn13 = None
+        book_cover = None
+
+        IDENTIFIER_TYPES = {'02': 'isbn10', '15': 'isbn13'}
+
+        found_identifiers = {}
+        for identifier in identifiers:
```

Comment:
this looks incomplete / untested. There likely needs to be an if statement which checks that `identifier[0].text` is in  `IDENTIFIER_TYPES`. In fact, you should include a test-case for an identifier type which isn't defined in `IDENTIFIER_TYPES` above. e.g. '04': 'whatever'